Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+            FileInfo editorMetaFile = assetDirectory.GetFiles("*sampleEditorDllMeta.txt", SearchOption.AllDirectories).FirstOrDefault();
+            if (editorMetaFile == null)
+            {
+                throw new FileNotFoundException("Could not find sample editor dll.meta file");
+            }
+            FileInfo uapMetaFile = assetDirectory.GetFiles("*sampleUAPDllMeta.txt", SearchOption.AllDirectories).FirstOrDefault();
+            if (uapMetaFile == null)
+            {
+                throw new FileNotFoundException("Could not find sample UAP dll.meta file");
+            }
+            FileInfo standaloneMetaFile = assetDirectory.GetFiles("*sampleStandaloneDllMeta.txt", SearchOption.AllDirectories).FirstOrDefault();
+            if (standaloneMetaFile == null)
+            {
+                throw new FileNotFoundException("Could not find sample Standalone dll.meta file");
+            }
+
+            string[] metaFileContent = File.ReadAllLines(editorMetaFile.FullName);
+
+            foreach (FileInfo dllFile in new DirectoryInfo(Application.dataPath.Replace("Assets", "NuGet/Plugins")).GetFiles("*", SearchOption.AllDirectories))
+            {
+                if (dllFile.Extension.Equals(".meta"))
+                {
+                    string dllFileName = dllFile.Name.Remove(dllFile.Name.Length - 5);
+                    if (dllGuids.ContainsKey(dllFileName))
+                    {
+                        metaFileContent[1] = string.Format("guid: {0}", dllGuids[dllFileName]);
+                        File.WriteAllLines(dllFile.FullName, metaFileContent);
+                    }
+                }
+            }
+
+            DirectoryInfo[] directories = new DirectoryInfo(Application.dataPath.Replace("Assets", "NuGet/Plugins")).GetDirectories("*", SearchOption.AllDirectories);
+            string dllGuid = string.Empty;
+            bool isUAP = false;
+            foreach (DirectoryInfo directory in directories)
+            {
+                isUAP = false;
+                if (!directory.Name.Equals("EditorPlayer"))
+                {
+                    if (directory.Name.Contains("Standalone") || directory.Parent.Name.Contains("Standalone"))
+                    {
+                        metaFileContent = File.ReadAllLines(standaloneMetaFile.FullName);
+                    }
+                    else if (directory.Name.Contains("UAP") || directory.Parent.Name.Contains("UAP"))
+                    {
+                        metaFileContent = File.ReadAllLines(uapMetaFile.FullName);
+                        isUAP = true;
+                    }
+
+                    FileInfo[] files = directory.GetFiles("*.dll", SearchOption.TopDirectoryOnly);
+                    string metaFilePath = string.Empty;
+                    foreach (FileInfo file in files)
+                    {
+                        //Editor is guid + 1
+                        //Standalone is guid + 2
```

Comment:
>//Standalone is guid + 2 [](start = 24, length = 24)

Is it the case that the editor guide + 1 case has already been done (i.e. it's already been +1ed before entering this function? My confusion comes from seeing comments that say there can be up to 3 cycles applied, but there's only two calls to CycleGuidForward here.

Reply:
Yes, it has been done when we copied editor dlls, will update.

---
In reply to: [277829124](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277829124) [](ancestors = 277829124)